app/javascript/packages/components/button-to.tsx

```diff
@@ -0,0 +1,49 @@
+import { useRef } from 'react';
+import { createPortal } from 'react-dom';
+import Button from './button';
+import type { ButtonProps } from './button';
+
+interface ButtonToProps extends ButtonProps {
```

Comment:
would we ever need to add params here that could be posted? or is that the kind of thing we'd add if/when we need it?

Reply:
> would we ever need to add params here that could be posted? or is that the kind of thing we'd add if/when we need it?

Possibly? I liken this to Rails' `button_to`, which (TIL) [supports a `params` hash](https://api.rubyonrails.org/classes/ActionView/Helpers/UrlHelper.html#method-i-button_to) for just that purpose. But I heavily subscribe to the school of YAGNI, so I'm not in any hurry to add it.

Reply:
YAGNI 👍